rest_framework-stubs/generics.pyi

```diff
@@ -19,12 +18,19 @@ def get_object_or_404(
 class UsesQuerySet(Protocol[_MT_co]):
     def get_queryset(self) -> QuerySet[_MT_co]: ...
 
+# Can't just use BaseFilterBackend because there's also things like django_filters.rest_framework.DjangoFilterBackend that are
+# valid options but don't extend it
+class BaseFilterProtocol(Protocol):
+    def filter_queryset(self, request: Request, queryset: QuerySet[_MT_co], view: views.APIView) -> QuerySet[_MT_co]: ...
+    def get_schema_fields(self, view: views.APIView) -> List[Any]: ...
+    def get_schema_operation_parameters(self, view: views.APIView): ...
+
 class GenericAPIView(views.APIView, UsesQuerySet[_MT_co]):
     queryset: Optional[Union[QuerySet[_MT_co], Manager[_MT_co]]] = ...
     serializer_class: Optional[Type[BaseSerializer]] = ...
     lookup_field: str = ...
     lookup_url_kwarg: Optional[str] = ...
-    filter_backends: Sequence[Type[BaseFilterBackend]] = ...
+    filter_backends: Sequence[Type[BaseFilterProtocol]] = ...
```

Comment:
Maybe a `Union[]`? It would be just more readable.

And please, add a test with something like `django-filter` has! 👍

Reply:
Can the test use `django-filter` even if the main package doesn't have a dep on it?

Reply:
We can create a custom class with the same signature.

Reply:
I've added a test for this, but it's failing for other reasons